pkg/secrets-store/stats_reporter.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secretsstore
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/metric"
+)
+
+var (
+	providerKey             = "provider"
+	errorKey                = "error_type"
+	namespaceKey            = "namespace"
+	nodePublishTotal        metric.Int64Counter
+	nodeUnPublishTotal      metric.Int64Counter
+	nodePublishErrorTotal   metric.Int64Counter
+	nodeUnPublishErrorTotal metric.Int64Counter
+	syncK8sSecretTotal      metric.Int64Counter
+	syncK8sSecretDuration   metric.Float64Measure
+)
+
+type reporter struct {
+	meter metric.Meter
+}
+
+type StatsReporter interface {
+	reportNodePublishCtMetric(provider string)
+	reportNodeUnPublishCtMetric()
+	reportNodePublishErrorCtMetric(provider, errType string)
+	reportNodeUnPublishErrorCtMetric()
+	reportSyncK8SecretCtMetric(namespace string, count int)
```

Comment:
I think namespace could be a sensitive data. instead, can we use provider for now?

Reply:
do we also want to report on runtime? windows vs linux?

Reply:
> I think namespace could be a sensitive data. instead, can we use provider for now?

Makes sense! we can start with provider and add more labels if required.

> do we also want to report on runtime? windows vs linux?

that'll be a good label to filter out errors. I'll add it.

Reply:
done!